Sources/Sentry/SentryScope.m

```diff
@@ -460,13 +460,16 @@ - (SentryEvent *__nullable)applyToEvent:(SentryEvent *)event
     @synchronized(_spanLock) {
         span = self.span;
     }
+    if (![event.type isEqualToString:SentryEnvelopeItemTypeTransaction])
+        if ([span isKindOfClass:[SentryTracer class]])
+            event.transaction = [(SentryTracer *)span name];
+    if (newContext == nil)
+        newContext = [NSMutableDictionary new];
+    newContext[@"trace"] = [span serialize];
```

Comment:
`m`: We could check if the span is `nil` first and only then create the `NSMutableDictionary` above. If the `span` is `nil` and the context is empty, we end up with an empty dictionary anyway.